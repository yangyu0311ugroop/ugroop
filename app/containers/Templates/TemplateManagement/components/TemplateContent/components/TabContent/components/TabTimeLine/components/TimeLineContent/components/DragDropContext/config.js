/**
 * Created by stephenkarpinskyj on 5/4/18.
 */

import {
  TEMPLATE_MANAGEMENT_DATASTORE,
  ON_DRAG_START,
  ON_DRAG_END,
} from 'appConstants';

export const CONFIG = {
  setValue: {
    [ON_DRAG_START]: [TEMPLATE_MANAGEMENT_DATASTORE, ON_DRAG_START],
    [ON_DRAG_END]: [TEMPLATE_MANAGEMENT_DATASTORE, ON_DRAG_END],
  },
};
