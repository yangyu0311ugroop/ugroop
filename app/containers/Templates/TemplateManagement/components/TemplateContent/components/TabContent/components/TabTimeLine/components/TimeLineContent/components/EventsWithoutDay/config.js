/**
 * Created by stephenkarpinskyj on 8/10/18.
 */

import { PORTAL_HELPERS } from 'containers/Portal/helpers';
import { EVENT_STORE_VIEW_SELECTORS } from 'datastore/eventStore/selectors';

export const CONFIG = {
  value: {},
  setValue: {
    eventCreate: EVENT_STORE_VIEW_SELECTORS.eventCreate,
    ...PORTAL_HELPERS.setValue,
  },
};
