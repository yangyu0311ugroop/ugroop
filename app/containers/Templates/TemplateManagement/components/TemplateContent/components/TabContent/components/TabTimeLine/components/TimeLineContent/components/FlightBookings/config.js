import { PORTAL_HELPERS } from 'containers/Portal/helpers';
import { EVENT_FLIGHT } from 'utils/modelConstants';
import {
  EVENT_STORE_DATA_SELECTORS,
  EVENT_STORE_VIEW_SELECTORS,
} from 'datastore/eventStore/selectors';
import { NODE_STORE_SELECTORS } from 'datastore/nodeStore/selectors';

// TODO: Start selecting from a well-maintained id array in template.calculated.flightBookingIds

export const CONFIG_1 = () => ({
  value: {
    flightBookingIds: EVENT_STORE_DATA_SELECTORS.flightBookingIds,
    eventIds: ({ templateId: id }) =>
      NODE_STORE_SELECTORS.calculatedNodesEvents({ id }),
  },
});

export const CONFIG_2 = () => ({
  value: {
    flightBookingIds: EVENT_STORE_DATA_SELECTORS.filterFlightBookingsByTemplateId(
      { idsProp: 'flightBookingIds' },
    ),
    flightIds: NODE_STORE_SELECTORS.filterByTypes({
      ids: 'eventIds',
      types: [EVENT_FLIGHT],
    }),
  },
  setValue: {
    flightBookingCreate: EVENT_STORE_VIEW_SELECTORS.flightBookingCreate,
    flightBookingView: EVENT_STORE_VIEW_SELECTORS.flightBookingView,
    ...PORTAL_HELPERS.setValue,
  },
});
