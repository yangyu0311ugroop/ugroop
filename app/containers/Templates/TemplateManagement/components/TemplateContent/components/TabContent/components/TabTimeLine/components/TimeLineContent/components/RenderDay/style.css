.container {
  background-color: #edf2f4;
}

.activity-container {
  border-left: 1px dashed #e5ecf0;
  padding-left: 30px;
}
