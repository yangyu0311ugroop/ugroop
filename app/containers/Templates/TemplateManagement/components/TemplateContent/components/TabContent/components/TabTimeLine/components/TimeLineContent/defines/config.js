import { GET_CHILDREN, NODE_API, UPDATE_CHILD } from 'apis/constants';

import { TEMPLATE_MANAGEMENT_VIEWSTORE } from 'appConstants';
import {
  NODE_STORE_SELECTORS,
  NODE_VIEW_STORE_SELECTORS,
} from 'datastore/nodeStore/selectors';

export const CONFIG = {
  value: {
    dayIds: NODE_STORE_SELECTORS.children,
    selectDayId: [TEMPLATE_MANAGEMENT_VIEWSTORE, 'selectedId'],
    editDayId: [TEMPLATE_MANAGEMENT_VIEWSTORE, 'editDayId'],
    editable: NODE_VIEW_STORE_SELECTORS.editable,
  },
  isLoading: {
    isUpdatingNode: [NODE_API, UPDATE_CHILD],
    isGettingChildren: [NODE_API, GET_CHILDREN],
  },
  setValue: {
    node: NODE_STORE_SELECTORS.nodes,
  },
};
