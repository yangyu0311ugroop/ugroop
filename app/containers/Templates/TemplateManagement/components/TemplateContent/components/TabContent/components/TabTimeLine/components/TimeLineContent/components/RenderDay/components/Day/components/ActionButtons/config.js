import {
  NODE_STORE,
  TEMPLATE_MANAGEMENT_DATASTORE,
  TEMPLATE_MANAGEMENT_VIEWSTORE,
} from 'appConstants';
import { PORTAL_HELPERS } from 'containers/Portal/helpers';
import { COMMENT } from 'containers/Templates/TemplateManagement/components/Comment/config';
import commonAttribute from 'containers/Templates/TemplateManagement/defines/commonAttribute';
import { EVENT_PATHS } from 'datastore/eventStore/constants';
import { EVENT_STORE_VIEW_SELECTORS } from 'datastore/eventStore/selectors';
import {
  NODE_STORE_SELECTORS,
  NODE_VIEW_STORE_SELECTORS,
} from 'datastore/nodeStore/selectors';
import { SET_VALUE } from 'ugcomponents/SnackBar/config';

export const TEMPLATE_ID_CONFIG = {
  value: {
    templateId: [TEMPLATE_MANAGEMENT_DATASTORE, 'id'],
  },
};

export const CONFIG = {
  value: {
    startDate: ({ templateId }) =>
      NODE_STORE_SELECTORS.startDate({ id: templateId }),
    displayDate: ({ templateId }) =>
      NODE_STORE_SELECTORS.displayDate({ id: templateId }),
    weekDay: ({ templateId }) =>
      NODE_STORE_SELECTORS.weekDay({ id: templateId }),
    title: NODE_STORE_SELECTORS.content,
    activityIds: NODE_STORE_SELECTORS.children,
    dateTitle: commonAttribute.formatDate,
    editable: NODE_VIEW_STORE_SELECTORS.editable,
    editing: NODE_STORE_SELECTORS.editing,
    firstChild: ({ tabId }) => NODE_STORE_SELECTORS.child(0)({ id: tabId }),
    nextSelectedId: ({ tabId, index }) => [
      NODE_STORE,
      'nodes',
      tabId,
      'children',
      index,
    ],
    prevSelectedId: ({ tabId, index }) => [
      NODE_STORE,
      'nodes',
      tabId,
      'children',
      index - 2,
    ],
  },
  setValue: {
    nodeId: [COMMENT, 'nodeId'],
    nodeStore: [COMMENT, 'nodeStore'],
    nodeData: [COMMENT, 'nodeData'],
    addChecklistParentId: [
      TEMPLATE_MANAGEMENT_VIEWSTORE,
      'addChecklistParentId',
    ],
    selectedId: [TEMPLATE_MANAGEMENT_VIEWSTORE, 'selectedId'],
    eventCreate: EVENT_STORE_VIEW_SELECTORS.eventCreate,
    formType: EVENT_STORE_VIEW_SELECTORS.eventFormProp({
      path: EVENT_PATHS.type,
    }),
    formSubtype: EVENT_STORE_VIEW_SELECTORS.eventFormProp({
      path: EVENT_PATHS.subtype,
    }),
    editing: NODE_STORE_SELECTORS.editing,
    ...SET_VALUE,
    ...PORTAL_HELPERS.setValue,
  },
};
