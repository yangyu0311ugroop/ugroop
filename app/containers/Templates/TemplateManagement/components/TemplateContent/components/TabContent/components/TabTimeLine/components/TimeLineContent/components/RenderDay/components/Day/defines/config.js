import { TEMPLATE_MANAGEMENT_VIEWSTORE } from 'appConstants';
import { COMMENT as DISCUSSION_DRAWER } from 'containers/Templates/TemplateManagement/components/Comment/config';
import {
  NODE_STORE_SELECTORS,
  NODE_VIEW_STORE_SELECTORS,
} from 'datastore/nodeStore/selectors';
import { FILE_STORE_SELECTORS } from 'datastore/fileStore/selectors';
import { SET_VALUE } from 'ugcomponents/SnackBar/config';

export const CONFIG = {
  value: {
    icon: props => [
      TEMPLATE_MANAGEMENT_VIEWSTORE,
      'editDays',
      props.dayId,
      'icon',
    ],
    placeId: props => [
      TEMPLATE_MANAGEMENT_VIEWSTORE,
      'editDays',
      props.dayId,
      'placeId',
    ],
    timeZoneId: props => [
      TEMPLATE_MANAGEMENT_VIEWSTORE,
      'editDays',
      props.dayId,
      'timeZoneId',
    ],
    discussionDrawerNodeId: [DISCUSSION_DRAWER, 'nodeId'],
    fk: props => FILE_STORE_SELECTORS.selectFileId({ id: props.dayPhotoId }),
    oldMetaInfo: props =>
      FILE_STORE_SELECTORS.templateMetaInfo({ id: props.dayPhotoId }),
    nodeChildren: props => NODE_STORE_SELECTORS.children({ id: props.dayId }),
    isNew: ({ dayId }) => [
      TEMPLATE_MANAGEMENT_VIEWSTORE,
      'editDays',
      dayId,
      'isNew',
    ],
    checklists: props =>
      NODE_STORE_SELECTORS.parentChecklists({ parentNodeId: props.dayId }),
    showChecklists: ({ dayId }) =>
      NODE_STORE_SELECTORS.calculatedShowChecklists({ id: dayId }),
    editing: ({ dayId }) => NODE_STORE_SELECTORS.editing({ id: dayId }),
    editable: NODE_VIEW_STORE_SELECTORS.editable,
    layout: ({ templateId }) =>
      NODE_STORE_SELECTORS.calculatedLayout({ id: templateId }),
  },

  setValue: {
    selectDayId: [TEMPLATE_MANAGEMENT_VIEWSTORE, 'selectedId'],
    selectedActivityId: [TEMPLATE_MANAGEMENT_VIEWSTORE, 'selectedActivityId'],
    editDayId: [TEMPLATE_MANAGEMENT_VIEWSTORE, 'editDayId'],
    editDays: [TEMPLATE_MANAGEMENT_VIEWSTORE, 'editDays'],
    editing: ({ dayId }) => NODE_STORE_SELECTORS.editing({ id: dayId }),
    ...SET_VALUE,
  },
};
