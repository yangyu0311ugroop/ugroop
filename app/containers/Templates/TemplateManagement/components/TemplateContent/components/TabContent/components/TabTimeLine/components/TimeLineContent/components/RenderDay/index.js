import { withStyles } from '@material-ui/core';
import { TEMPLATE_ID_CONFIG } from 'apis/components/Node/config';
import TabCustom from 'containers/Templates/TemplateManagement/components/TemplateContent/components/TabContent/components/TabCustom';
import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';
import { compose } from 'redux';
import resaga from 'resaga';
import Node from 'smartComponents/Node';
import { VARIANTS } from 'variantsConstants';
import Day from './components/Day';
import { CONFIG } from './defines/dayConfig';

const styles = {
  marginTop: {
    marginTop: 16,
  },

  checklistDay: {
    position: 'absolute',
    left: -47,
    top: 47,
  },

  relative: {
    position: 'relative',
  },

  checklists: {
    marginBottom: 8,
    border: '1px solid rgba(164, 172, 186, 0.5)',
    borderRadius: 4,
    overflow: 'hidden',
  },
};

export class RenderDay extends PureComponent {
  componentWillReceiveProps = nextProps => {
    const { dayId } = this.props;

    if (dayId !== nextProps.dayId) {
      this.props.resaga.setValue({
        editing: false,
      });
    }
  };

  componentWillUnmount = () => {
    const { editing } = this.props;

    if (editing) {
      this.props.resaga.setValue({
        editing: false,
      });
    }
  };

  canEdit = () => {
    const { editing } = this.props;

    return editing;
  };

  renderChildren = () => {
    const { classes, dayId, activityIds, isPublic } = this.props;

    if (activityIds && activityIds.length > 0) {
      return (
        <div className={classes.marginTop}>
          <TabCustom
            id={dayId}
            offset
            elevation={1}
            editing={this.canEdit()}
            isPublic={isPublic}
          />
        </div>
      );
    }

    // no more sections placeholder
    return null;
  };

  renderDay = () => {
    const {
      id,
      dayId,
      index,
      activityIds,
      photoId,
      hasContent,
      content,
      url,
      description,
      tabId,
      showEventDetail,
      marginLeft,
      isPublic,
      editing,
    } = this.props;

    let hasSections = false;
    if (activityIds && activityIds.length > 0) {
      hasSections = true;
    }

    return (
      <Day
        templateId={id}
        tabId={tabId}
        selected={this.props.selected}
        index={index}
        dayPhotoId={photoId[0]}
        dayId={dayId}
        onDelete={this.handleDelete}
        hasContent={hasContent}
        content={content}
        url={url}
        description={description}
        hasSections={hasSections}
        willBeEdited={this.props.willBeEdited}
        showEventDetail={showEventDetail}
        marginLeft={marginLeft}
        isPublic={isPublic}
        editing={editing}
      >
        {this.renderChildren()}
      </Day>
    );
  };

  render = () => {
    const { dayId } = this.props;
    return (
      <>
        {this.renderDay()}
        <Node id={dayId} key={dayId} variant={VARIANTS.LOGIC} />
      </>
    );
  };
}

RenderDay.propTypes = {
  resaga: PropTypes.object.isRequired,

  // hoc
  dayId: PropTypes.number.isRequired,
  index: PropTypes.number,
  selected: PropTypes.bool,
  willBeEdited: PropTypes.bool,
  showEventDetail: PropTypes.bool,
  marginLeft: PropTypes.bool,
  tabId: PropTypes.number,
  classes: PropTypes.object,

  // Parent
  isPublic: PropTypes.bool,

  // resaga
  id: PropTypes.number, // template id
  activityIds: PropTypes.array,
  photoId: PropTypes.any,
  hasContent: PropTypes.bool,
  editing: PropTypes.bool,
  content: PropTypes.string,
  url: PropTypes.string,
  description: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
};

RenderDay.defaultProps = {
  activityIds: [],
  photoId: [],
  willBeEdited: false,
};

export default compose(
  withStyles(styles, { name: 'RenderDay' }),
  resaga(TEMPLATE_ID_CONFIG),
  resaga(CONFIG),
)(RenderDay);
