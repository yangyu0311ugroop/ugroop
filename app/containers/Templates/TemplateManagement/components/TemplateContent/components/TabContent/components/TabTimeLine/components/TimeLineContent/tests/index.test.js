import { DEFAULT_SELECTDAY_INDEX } from 'containers/Templates/TemplateManagement/components/TemplateContent/components/TabContent/components/TabTimeLine/config';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { List } from 'immutable';
/**
 * Created by quando on 3/9/17.
 */
import React from 'react';
import RenderDay from '../components/RenderDay';
import { TimeLineContent } from '../index';

jest.mock('modernizr', () => 'mock');

describe('TimeLineContent/tests/index.test.js', () => {
  const id = 2;
  const ids = [1, 2];
  const tab = { parentNodeId: 1 };
  const parentData = { hi: 'ho' };

  const resagaMock = {
    analyse: jest.fn(),
    dispatch: jest.fn(),
    dispatchTo: jest.fn(),
    setValue: jest.fn(),
    getValue: jest.fn(() => List()),
    isLoading: jest.fn(),
  };
  const props = {
    id,
    tab,
    classes: {},
    resaga: resagaMock,
  };
  let rendered;
  let instance;

  const dayData = {};

  beforeEach(() => {
    global.setTimeout = jest.fn(cb => cb());
    rendered = shallow(
      <TimeLineContent dayIds={ids} dayData={dayData} {...props} />,
    );
    instance = rendered.instance();
  });
  afterEach(() => jest.clearAllMocks());

  describe('<TimeLineContent />', () => {
    it('should exists', () => {
      expect(TimeLineContent).toBeDefined();
    });
    it('should render without exploding', () => {
      expect(rendered.length).toBe(1);
    });
    it('should render error without exploding', () => {
      rendered.setState({ error: 'Error' });
      expect(rendered.length).toBe(1);
    });
    it('should render without exploding with selectDayId unset', () => {
      rendered = shallow(
        <TimeLineContent
          {...props}
          dayIds={ids}
          dayData={dayData}
          parentData={parentData}
          selectDayId={1}
        />,
      );
      expect(rendered.length).toBe(1);
    });
  });

  describe('<TimeLineContent /> props', () => {
    it('RenderDay', () => {
      expect(rendered.find(RenderDay).length).toBe(ids.length);
    });
    it('no day data', () => {
      rendered = shallow(
        <TimeLineContent
          parentData={parentData}
          dayData={dayData}
          {...props}
        />,
      );
      expect(toJSON(rendered)).toMatchSnapshot();
    });
  });

  describe('<TimeLineContent /> methods: ', () => {
    describe('onCreate() ', () => {
      it('should call proper data', () => {
        instance.onCreate();
        expect(resagaMock.dispatchTo).toBeCalled();
      });
    });
  });
  describe('onSuccessUpdate() ', () => {
    it('should call proper data', () => {
      const data = { payload: { id: 1 } };
      instance.onSuccessUpdate(data);
      expect(resagaMock.dispatchTo).toBeCalled();
    });
  });

  describe('render()', () => {
    it('should render LoadingText', () => {
      rendered.setProps({ dayIds: [], fetching: true });

      const snapshot = shallow(<div>{instance.render()}</div>);

      expect(toJSON(snapshot)).toMatchSnapshot();
    });

    it('should render DEFAULT_SELECTDAY_INDEX', () => {
      rendered.setProps({
        dayIds: [1],
        editable: true,
        selectDayId: DEFAULT_SELECTDAY_INDEX,
      });

      const snapshot = shallow(<div>{instance.render()}</div>);

      expect(toJSON(snapshot)).toMatchSnapshot();
    });

    it('should render editable', () => {
      rendered.setProps({ dayIds: [], editable: true });

      const snapshot = shallow(<div>{instance.render()}</div>);

      expect(toJSON(snapshot)).toMatchSnapshot();
    });
  });
});
