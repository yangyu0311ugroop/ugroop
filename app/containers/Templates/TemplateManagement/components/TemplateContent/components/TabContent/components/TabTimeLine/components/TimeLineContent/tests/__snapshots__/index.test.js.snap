// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeLineContent/tests/index.test.js <TimeLineContent /> props no day data 1`] = `
<WithStyles(GridContainer)
  card={true}
  direction="column"
>
  <WithStyles(GridItem)>
    <WithStyles(GridContainer)
      alignItems="stretch"
      justify="space-between"
      spacing={0}
    >
      <div>
        <WithStyles(Icon)
          icon="lnr-calendar-empty"
          size="xl"
        />
        <WithStyles(H3)>
          <FormattedMessage
            defaultMessage="There are no Days in this Tour"
            id="app.containers.Template.TimeLineContent.NoDayTitlePlaceHolder"
          />
        </WithStyles(H3)>
      </div>
    </WithStyles(GridContainer)>
  </WithStyles(GridItem)>
</WithStyles(GridContainer)>
`;

exports[`TimeLineContent/tests/index.test.js render() should render DEFAULT_SELECTDAY_INDEX 1`] = `
<div>
  <ConnectFunction>
    <WithStyles(GridContainer)
      direction="column"
      spacing={0}
    >
      <WithStyles(Connect(WithSimplifyResaga(Connect(WithSimplifyResaga(RenderDay)))))
        dayId={1}
        index={0}
        key="1"
        tabId={2}
        willBeEdited={false}
      />
      <WithStyles(GridItem)>
        <WithStyles(GridContainer)
          alignItems="center"
          direction="column"
          spacing={0}
        >
          <WithStyles(GridItem)>
            <WithStyles(JTextClass)
              bold={true}
              gray={true}
              sm={true}
              spacing2={true}
              uppercase={true}
            >
              End Of Itinerary
            </WithStyles(JTextClass)>
          </WithStyles(GridItem)>
        </WithStyles(GridContainer)>
      </WithStyles(GridItem)>
    </WithStyles(GridContainer)>
  </ConnectFunction>
</div>
`;

exports[`TimeLineContent/tests/index.test.js render() should render LoadingText 1`] = `
<div>
  <WithStyles(GridContainer)
    card={true}
    direction="column"
  >
    <WithStyles(GridItem)>
      <WithStyles(LoadingText) />
    </WithStyles(GridItem)>
  </WithStyles(GridContainer)>
</div>
`;

exports[`TimeLineContent/tests/index.test.js render() should render editable 1`] = `
<div>
  <WithStyles(GridContainer)
    card={true}
    direction="column"
  >
    <WithStyles(GridItem)>
      <WithStyles(GridContainer)
        alignItems="stretch"
        justify="space-between"
        spacing={0}
      >
        <div>
          <WithStyles(Icon)
            icon="lnr-calendar-empty"
            size="xl"
          />
          <WithStyles(H3)>
            <FormattedMessage
              defaultMessage="There are no Days in this Tour"
              id="app.containers.Template.TimeLineContent.NoDayTitlePlaceHolder"
            />
          </WithStyles(H3)>
          <WithStyles(H5)>
            <FormattedMessage
              defaultMessage="Use \\"Add Day\\" to begin building the tour."
              id="app.containers.Template.TimeLineContent.NoDayDescriptionPlaceHolder"
            />
          </WithStyles(H5)>
          <WithStyles(Button)
            color="green"
            disabled={true}
            onClick={[Function]}
            size="small"
          >
            Add Day
          </WithStyles(Button)>
        </div>
      </WithStyles(GridContainer)>
    </WithStyles(GridItem)>
  </WithStyles(GridContainer)>
</div>
`;
